src/modules/user-adapter/user.repository.ts

```diff
@@ -0,0 +1,23 @@
+import { Logger } from '@nestjs/common';
+import { User, UserRepository } from '../../packages/user/user.types';
+import fetch from 'node-fetch';
+
+export class RestUserRepository implements UserRepository {
+    constructor(userAdapterUrl: string) {
+        this.userAdapterUrl = userAdapterUrl;
+    }
+    private readonly userAdapterUrl: string;
+    private readonly logger = new Logger(RestUserRepository.name);
+
+    public async getCurrentUser(header: string): Promise<User> {
+        try {
+            const response = await fetch(this.userAdapterUrl, {
+                headers: [['authorization', header]],
+            });
+            return await response.json();
+        } catch (error) {
+            this.logger.error(error);
+            throw error;
```

Comment:
I'm not sure we need the try catch here. Do we?

Reply:
this is so we can log errors that are thrown in the fetch command